internal/operands/vm-delete-protection/reconcile.go

```diff
@@ -0,0 +1,98 @@
+package vm_delete_protection
+
+import (
+	"fmt"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Define RBAC rules needed by this operand:
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingadmissionpolicies;validatingadmissionpolicybindings,verbs=get;list;create;watch;update;delete
+
+const (
+	operandName                              = "vm-delete-protection"
+	operandComponent                         = common.AppComponentVMDeletionProtection
+	virtualMachineDeleteProtectionPolicyName = "kubevirt-vm-deletion-protection"
+)
+
+func init() {
+	utilruntime.Must(admissionregistrationv1.AddToScheme(common.Scheme))
+}
+
+func WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &admissionregistrationv1.ValidatingAdmissionPolicy{}},
+		{Object: &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}},
+	}
+}
+
+type VMDeleteProtection struct{}
+
+var _ operands.Operand = &VMDeleteProtection{}
+
+func New() operands.Operand {
+	if err := checkCelExpression(); err != nil {
+		panic(fmt.Errorf("invalid VM delete protection CEL expression %v", err))
```

Comment:
We should be careful with panicking. Also, this adds a new import for a rather complex package. Are we sure we need that?

IMO we are better of failing the reconciliation when creating/updating a VAP is rejected by the admitter. I also see `vap.Status.TypeChecking` being evaluated below.

Reply:
Maybe we can keep the unit test performed over this expression, and drop the panic. In this way, we will have some kind of feedback in case the expression would become incorrect (maybe because something in CEL syntax is changed or deprecated).

Reply:
I think this is a good use of panic, because it is a programmer's error.  We can move this check to a unit test.

The package is already imported, by another dependency. We were depending on it even before this PR.

Reply:
Adjusted.

Reply:
I see, it was an indirect import before. Still, I think it makes more sense to simply verify deployed VAPs work as expected.